src/databricks/labs/ucx/hive_metastore/federation.py

```diff
@@ -31,58 +48,161 @@ def enable(self):
+    }
+
+    def create_from_cli(self, prompts: Prompts):
         if not self._enable_hms_federation:
             raise RuntimeWarning('Run `databricks labs ucx enable-hms-federation` to enable HMS Federation')
         name = self._workspace_info.current()
-        connection_info = self._get_or_create_connection(name)
+
+        ext_hms = None
+        try:
+            ext_hms = self._get_ext_hms()
+        except ValueError:
+            logger.info('Failed to retrieve external Hive Metastore connection information')
+
+        if ext_hms and prompts.confirm(
+            f'A supported external Hive Metastore connection was identified: {ext_hms.db_type}. Use this connection?'
+        ):
+            connection_info = self._get_or_create_ext_connection(name, ext_hms)
+        else:
+            connection_info = self._get_or_create_int_connection(name)
         assert connection_info.name is not None
+        return self._register_federated_catalog(connection_info)
+
+    def _get_ext_hms(self) -> ExtHms:
+        config = self._installation.load(WorkspaceConfig)
+        if not config.spark_conf:
+            raise ValueError('Spark config not found')
+        spark_config = config.spark_conf
+        jdbc_url = self._get_value_from_config_key(spark_config, 'spark.hadoop.javax.jdo.option.ConnectionURL')
+        if not jdbc_url:
+            raise ValueError('JDBC URL not found')
+        version = self._get_value_from_config_key(spark_config, 'spark.sql.hive.metastore.version')
+        # extract major version from version using regex
+        if not version:
+            raise ValueError('Hive Metastore version not found')
+        major_version = re.match(r'(\d+\.\d+)', version)
+        if not major_version:
+            raise ValueError(f'Invalid Hive Metastore version: {version}')
+        version = major_version.group(1)
+        ext_hms = replace(self._split_jdbc_url(jdbc_url), version=version)
+        supported_versions = self.supported_db_vers.get(ext_hms.db_type)
+        if not supported_versions:
+            raise ValueError(f'Unsupported Hive Metastore: {ext_hms.db_type}')
+        if version not in supported_versions:
+            raise ValueError(f'Unsupported Hive Metastore Version: {ext_hms.db_type} - {version}')
+        if not ext_hms.user:
+            ext_hms = replace(
+                ext_hms,
+                user=self._get_value_from_config_key(spark_config, 'spark.hadoop.javax.jdo.option.ConnectionUserName'),
+            )
+        if not ext_hms.password:
+            ext_hms = replace(
+                ext_hms,
+                password=self._get_value_from_config_key(
+                    spark_config, 'spark.hadoop.javax.jdo.option.ConnectionPassword'
+                ),
+            )
+        return ext_hms
+
+    @classmethod
+    def _split_jdbc_url(cls, jdbc_url: str) -> ExtHms:
+        # Define the regex pattern to match the JDBC URL components
+        pattern = re.compile(
```

Comment:
Use urlparse here:

```
from urllib.parse import urlparse, ParseResult
```

see LocationTrie

Reply:
It doesn't work well with JDBC URLs